src/core/Form/Textarea/Textarea.tsx

```diff
@@ -1,52 +1,161 @@
-import React, { Component } from 'react';
+import React, { Component, ChangeEvent, FocusEvent } from 'react';
 import { default as styled } from 'styled-components';
+import classnames from 'classnames';
 import { TokensProp, InternalTokensProp } from '../../theme';
-import {
-  Textarea as CompTextarea,
-  TextareaProps as CompTextareaProps,
-} from '../../../components/Form/Textarea';
 import { baseStyles } from './Textarea.baseStyles';
 import { withSuomifiDefaultProps } from '../../theme/utils';
-import classnames from 'classnames';
+import { HtmlTextarea, HtmlTextareaProps, HtmlDiv } from '../../../reset';
+import { AutoId } from '../../../utils/AutoId';
+import { LabelText } from '../LabelText/LabelText';
+import { HintText } from '../HintText/HintText';
+import { StatusText } from '../StatusText/StatusText';
+import { InputStatus } from '../types';
 
 const baseClassName = 'fi-textarea';
-
 const textareaClassNames = {
+  textareaContainer: `${baseClassName}_textarea-element-container`,
+  textarea: `${baseClassName}_textarea`,
   disabled: `${baseClassName}--disabled`,
   error: `${baseClassName}--error`,
 };
 
-export interface TextareaProps extends CompTextareaProps, TokensProp {}
+type TextareaStatus = Exclude<InputStatus, 'success'>;
```

Comment:
Now only TextInput is having all statuses: **_default, error and success_**

Textarea and Checkbox is excluding the **success** out from the InputStatus. Should we change it other way around as it seems that only TextInput will use the success?

https://github.com/vrk-kpa/suomifi-ui-components/blob/de3ba041127f51eb2df5c75ff152933eefc16e13/src/core/Form/Checkbox/Checkbox.tsx#L37

Reply:
I wouldn't change is as it is a part of status text component functionality. Changing it would lead to "customization" and move parts of implementation (styles) outside the component.